### DISCUSSION TITLE
Using LoggingTriggerHistoryPlugin to save trigger history in DB in AspNetCore

### DISCUSSION THREAD

**Discussion by mniyad** [quartznet/quartznet#1295] [Q&A]
Hi
I'm trying to save trigger history in my job store (SQL Server).  I'm hosting scheduler server in AspNetCore and I can't seem to find an example on how to use the LoggingTriggerHistoryPlugin plugin to achieve this goal.  There're a couple out there using name-value pair inputs to StdSchedulerFactory. However, I'm setting it up in code.

Any pointers to achieve this welcome.

Thanks

**Comment by lahma** [comment]
I think support for this specific plugin is currently missing from strongly-typed API, but you could check this for an [example how to extend configuration](https://github.com/quartznet/quartznet/blob/main/src/Quartz.Plugins/PluginConfigurationExtensions.cs). Note that you can always also supply the configuration keys like shown in samples. 

If you come up with extension methods that improve the code-based configuration story, a pull request would be greatly appreciated!

**Reply by mniyad** [reply]
Thanks Mark
While digging in the codebase I found the following folder

[Quartz.Web/History](https://github.com/quartznet/quartznet/tree/main/src/Quartz.Web/History)

This is very similar to what I'm trying to do.

On a side note, Quartz.Web looks like a webapi for Quartz.net? What happened to it? Looks abandoned, has it moved elsewhere?

**Reply by lahma** [reply]
The web API is in hibernation, but [there's an issue about it.](https://github.com/quartznet/quartznet/issues/511)
